### PR TITLE
Fix Goroutine leaks in several packages

### PR DESCRIPTION
**Pull request by akagami-harsh**
## Which problem is this PR solving?
- fix gorourine leak in some packages 

## Description of the changes
- part of #5006 

## How was this change tested?
- go test

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`

### CODE REVIEW

**Comment by akagami-harsh** [comment]
I've ignored `goleak` check for `github.com/golang/glog.(*fileSink).flushDaemon` in certain packages, as is couldn't find a fix for it

File: cmd/internal/env/command_test.go [hunk 1]
```diff
@@ -30,3 +31,7 @@ func TestCommand(t *testing.T) {
 	assert.True(t, strings.Contains(buf.String(), "METRICS_BACKEND"))
 	assert.True(t, strings.Contains(buf.String(), "SPAN_STORAGE"))
 }
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m, goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon"))
```
Comment (yurishkuro) [inline cmd/internal/env/command_test.go:36]: > goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon")

Let's move this to pkg/testutils (with comment explaining why it's needed) and reuse instead of repeating everywhere.

File: cmd/agent/app/reporter/grpc/builder.go [hunk 2]
```diff
@@ -102,6 +102,9 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger, mFactory metrics.Fact
 	dialOptions = append(dialOptions, grpc.WithUnaryInterceptor(grpc_retry.UnaryClientInterceptor(grpc_retry.WithMax(b.MaxRetry))))
 	dialOptions = append(dialOptions, b.AdditionalDialOptions...)
 
+	connCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
```
Comment (yurishkuro) [inline cmd/agent/app/reporter/grpc/builder.go:106]: `CreateConnection` is not a blocking function, it will exit after setting things up and this defer will be executed. Is that what you intended?

Please move this specific change to a different PR, as other packages don't require substantial changes and the rest of this PR can be merged.

File: cmd/internal/env/command_test.go (hunk 1 above)
Comment (akagami-harsh) [reply cmd/internal/env/command_test.go:36]: yes, that would be better

File: cmd/agent/app/reporter/grpc/builder.go (hunk 2 above)
Comment (akagami-harsh) [reply cmd/agent/app/reporter/grpc/builder.go:106]: >CreateConnection is not a blocking function, it will exit after setting things up and this defer will be executed. Is that what you intended?

ohh my mistake, i should have just returned the `cancel()` function

Comment (akagami-harsh) [reply cmd/agent/app/reporter/grpc/builder.go:106]: >Please move this specific change to a different PR, as other packages don't require substantial changes and the rest of this PR can be merged.

Okay!

File: cmd/internal/env/command_test.go [hunk 3]
```diff
@@ -19,7 +19,9 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/jaegertracing/jaeger/pkg/testutils"
 	"github.com/stretchr/testify/assert"
+	"go.uber.org/goleak"
```
Comment (yurishkuro) [inline cmd/internal/env/command_test.go:24]: run `make fmt`

File: pkg/testutils/leakcheck.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	"go.uber.org/goleak"
+)
+
+// IgnoreGlogFlushDaemonLeak returns a goleak.Option that ignores the flushDaemon function
+// from the glog package that can cause false positives in leak detection.
+// This is necessary because glog starts a goroutine in the background that may not
+// be stopped when the test finishes, leading to a detected but expected leak.
+func IgnoreGlogFlushDaemonLeak() goleak.Option {
+	return goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon")
+}
```
Comment (yurishkuro) [inline pkg/testutils/leakcheck.go:13]: add package_test invoking this function to appease code coverage

File: pkg/testutils/leakcheck.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package testutils
+
+import (
+	"go.uber.org/goleak"
+)
+
+// IgnoreGlogFlushDaemonLeak returns a goleak.Option that ignores the flushDaemon function
+// from the glog package that can cause false positives in leak detection.
+// This is necessary because glog starts a goroutine in the background that may not
+// be stopped when the test finishes, leading to a detected but expected leak.
```
Comment (yurishkuro) [inline pkg/testutils/leakcheck.go:24]: great explanation!

File: pkg/testutils/leakcheck.go (hunk 4 above)
Comment (yurishkuro) [reply pkg/testutils/leakcheck.go:27]: Also, I just realized, we could've done this differently by adding `testutils.CheckGoroutineLeaks` function that would be called from the rest of the codebase, instead of having two dependencies everywhere (goleak and testutils).

**Review by yurishkuro — APPROVED** [review]

File: pkg/testutils/leakcheck.go (hunk 4 above)
Comment (akagami-harsh) [reply pkg/testutils/leakcheck.go:27]: yeah, we can also do that.

**Comment by yurishkuro** [comment]
Thanks!
